### PR TITLE
Allow user to complete equality and diversity monitoring task

### PR DESCRIPTION
**Pull request by patrickjfl**
This PR adds the remaining E&D task pages: military veteran, care leaver, carer responsibilities and marital status.

Screenshot 2023-08-16 at 09 08 15

Screenshot 2023-08-16 at 09 08 36

Screenshot 2023-08-16 at 09 08 50

Screenshot 2023-08-16 at 09 09 07

### CODE REVIEW

**Comment by patrickjfl** [comment]
[Equality & Diversity - Military Veteran](https://trello.com/c/L8sOLUxT/400-equality-diversity-military-veteran)

**Comment by patrickjfl** [comment]
[Equality & Diversity - Care leaver](https://trello.com/c/uojiX6tM/401-equality-diversity-care-leaver)

**Comment by patrickjfl** [comment]
[Equality & Diversity - Parental or carer responsibilities](https://trello.com/c/0mMHauSc/402-equality-diversity-parental-or-carer-responsibilities)

**Comment by patrickjfl** [comment]
[Equality & Diversity - Marital Status](https://trello.com/c/ZEGGtQ55/403-equality-diversity-marital-status)

File: server/@types/ui/index.d.ts [hunk 1]
```diff
@@ -29,6 +29,8 @@ export type YesOrNo = 'yes' | 'no'
 
 export type YesOrNoOrPreferNotToSay = 'yes' | 'no' | 'preferNotToSay'
 
+export type YesNoOrIDK = YesOrNo | 'iDontKnow'
+
```
Comment (edavey) [inline server/@types/ui/index.d.ts:33]: I wonder if the addition of this type -- which is going to be used in several pages -- might be worth its own commit? Or at least a mention in the commit message?

File: integration_tests/tests/apply/about_the_person/equality_and_diversity/complete_marital_status.cy.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
+//  Feature: Referrer completes 'marital status' question page
+//    So that I can complete the 'Equality questions' task
+//    As a referrer
+//    I want to answer questions on the marital status page
+//
+//  Scenario: submits a valid answer to marital status page
+//    Given I'm on the 'marital status' question page
+//    When I give a valid answer
+//    Then I am taken to the task list page page
```
Comment (edavey) [inline integration_tests/tests/apply/about_the_person/equality_and_diversity/complete_marital_status.cy.ts:9]: Should we also now (finally) note that the "task" is complete? e.g.

`// And I see that task is complete`

**Review by edavey — APPROVED** [review]
Great stuff, very easy to follow and will be a handy reference going forward.

Just a couple of tiny comments / suggestions...

File: server/@types/ui/index.d.ts (hunk 1 above)
Comment (patrickjfl) [reply server/@types/ui/index.d.ts:33]: yes, good idea. I'll sort that now

File: integration_tests/tests/apply/about_the_person/equality_and_diversity/complete_marital_status.cy.ts (hunk 2 above)
Comment (patrickjfl) [reply integration_tests/tests/apply/about_the_person/equality_and_diversity/complete_marital_status.cy.ts:9]: thanks, yes absolutely. I'm on it

**Review by Harriethw — APPROVED** [review]
